pkg/controller/schedule_controller.go

```diff
@@ -290,18 +311,26 @@ func parseCronSchedule(itm *api.Schedule) (cron.Schedule, []string) {
 		return nil, validationErrors
 	}
 
+	logContext := logger.WithField("schedule", kubeutil.NamespaceAndName(itm))
+
 	// adding a recover() around cron.Parse because it panics on empty string and is possible
 	// that it panics under other scenarios as well.
 	func() {
 		defer func() {
 			if r := recover(); r != nil {
-				glog.V(4).Infof("panic parsing schedule %v/%v, cron schedule=%v: %v", itm.Namespace, itm.Name, itm.Spec.Schedule, r)
+				logContext.WithFields(logrus.Fields{
+					"schedule": itm.Spec.Schedule,
+					"recover":  r,
+				}).Info("Panic parsing schedule")
 				validationErrors = append(validationErrors, fmt.Sprintf("invalid schedule: %v", r))
 			}
 		}()
 
 		if res, err := cron.ParseStandard(itm.Spec.Schedule); err != nil {
-			glog.V(4).Infof("error parsing schedule %v/%v, cron schedule=%v: %v", itm.Namespace, itm.Name, itm.Spec.Schedule, err)
+			logContext.WithFields(logrus.Fields{
+				"schedule": itm.Spec.Schedule,
+				"error":    err,
+			}).Info("Error parsing schedule")
 			validationErrors = append(validationErrors, fmt.Sprintf("invalid schedule: %v", err))
```

Comment:
see above about maybe using something like pkg/errors to attach context to the error message, rather than using logging.